orchestrator/graphql/authentication.py

```diff
@@ -0,0 +1,117 @@
+# Copyright 2022-2023 SURF.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Callable
+
+import strawberry
+import structlog
+from fastapi import HTTPException
+from graphql.pyutils import Path
+from strawberry import BasePermission
+from strawberry.types import Info
+from strawberry.types.fields.resolver import StrawberryResolver
+
+from orchestrator.settings import app_settings, oauth2_settings
+
+logger = structlog.get_logger(__name__)
+
+# Note: if this becomes generic enough we might want to move this to oauth2_lib, so it can be reused also by
```

Comment:
Maybe take this note into account and already move it to oauth2_lib?

Reply:
I doesn't take much time and all other projects using this class will benefit from it.

Reply:
It takes longer if people want tests for the authentication and don't want to keep this hanging, so it will be done in a different PR

Reply:
> people want tests for the authentication

🙋‍♂️